emission/itinerum/itinerum_boolean.py

```diff
@@ -0,0 +1,49 @@
+# Standard imports 
+import logging
+
+# Our imports
+import emission.storage.pipeline_queries as epq
+import emission.storage.timeseries.abstract_timeseries as esta
+import emission.storage.decorations.analysis_timeseries_queries as esda
+import emission.core.wrapper.itinerum_boolean as ecwib
+
+
+def set_itinerum_boolean(user_id):
+    time_query = epq.get_time_range_for_itinerum_boolean(user_id)
+    try: 
+        ibp = ItinerumBooleanPipeline()
+        ibp.user_id = user_id
+        ibp.run_itinerum_boolean_pipeline(user_id, time_query)
+        if ibp.last_trip_done is None:
+            logging.debug("After run, last_trip_done == None, must be early return")
+        epq.mark_itinerum_boolean_done(user_id, ibp.last_trip_done)
+    except:
+        logging.exception("Error while setting boolean, timestamp unchanged")
+        epq.mark_itinerum_boolean_failed(user_id)
+
+class ItinerumBooleanPipeline:
+    def __init__(self):
+        self.last_trip_done = None
+
+    @property
+    def last_trip_done(self):
+        return self._last_trip_done
+
+    def run_itinerum_boolean_pipeline(self, user_id, time_range):
+        #This is where we need to set the boolean?? 
+        #How is this done? 
+        #1.) seems to need to set the time series 
+        self.ts = esta.TimeSeries.get_time_series(user_id)
+        self.toSetFalseTrips = esda.get_entries(
+                esda.ITINERUM_BOOLEAN_KEY, user_id, time_query=time_range)
+        for trip in self.toSetFalseTrips:
+            itinerum_boolean = False
+            ib = ecwib.Itinerum_Boolean()
+            ib.trip_id = trip.get_id()
+            ib.itinerum_received = itinerum_boolean
+            ib.start_ts = trip.data.start_ts
+            ib.end_ts = trip.data.end_ts
+            #Insert into the database as the boolean "false"
+            self.ts.insert_data(self.user_id, esda.ITINERUM_BOOLEAN, ib)
+            if self._last_trip_done is None or self._last_trip_done.data.end_ts < trip.data.end_ts:
+                self._last_trip_done = trip
```

Comment:
@jruzekowicz @kafitz this is the key of what needs to be changed. The implementation here should be pretty straightforward, read all the data (trips, sections, points, etc) and store into a file labeled with the time range. No need to work with `esda.ITINERUM_BOOLEAN` which doesn't even exist

Reply:
To store the geojson output, look at https://github.com/e-mission/e-mission-server/blob/2177582fdea9a52faa8c1a238e62b111dbbcedba/emission/analysis/plotting/geojson/geojson_feature_converter.py

you can generate geojson for each trip instead of reading all the data. I would experiment with saving different kinds of outputs

Reply:
Currently looking at the extract_timeline_for_day_range_and_user file. Starting with the main function, it begins the parser and a mutually exclusive group. Is this something I should also incorporate? My understanding of this part of the code is that it sets up an args object to get alist of uuid's. However, we appear to be working with a single uuid within the run_export_data_pipeline. Thus, I am assuming this will be slightly different.

Reply:
It then calls export_timeline_for_users, passing in the uuid list and args. 
```
export_timeline(user_id=curr_uuid, start_day_str=args.start_day,
                end_day_str= args.end_day, timezone=args.timezone,
                file_name=args.file_prefix)
```
The main  function is this export_timeline section that pulls from args to get eh required variables. This is my thought process for needing the args object. 
However, in my run_export_data_pipeline, it passes in self, user_id, and time_range. I am trying to figure out the best way to build from here.

Reply:
@jruzekowicz the code pointers that I provide to you are intended to give you pointers that you can adapt to your needs. They are not intended for you to be able to copy-paste directly. You need to understand the code, figure out which parts of it are relevant, and use it appropriately.

e.g. look up what is a parser (https://duckduckgo.com/?t=ffsb&q=python+argparse&ia=web). Is it relevant to what you are doing?

Reply:
Remember from a high level, there will be some sort of scheduler process that invokes your script periodically. This probably won't be supplying any arguments itself. If argparse is responsible for taking in UUIDs from the CLI, there will be some other way our script will have to get these variables. 

Perhaps it's even easiest to hardcode a list now until it can be generated, depends on where you want to start.

Reply:
@kafitz this is currently structured as an additional stage of the intake pipeline, 
https://github.com/e-mission/e-mission-server/pull/823/files#diff-694dff3ba8c686137cb4726369a03b612728562e37fde55292b887269153fdaeR176

so it would be invoked as part of the regular intake script

To recap:
- there is a **new stage** which will be invoked as part of the intake pipeline
   - it will export values to a file
   - it will get its inputs from the intake pipeline and the pipeline state
- there is an **existing** script designed for one-time usage which exports **all** data
    - it gets its inputs from argparse
    - its export code can be reused/adapted for the new stage
    - however it cannot be directly copy-pasted

My question/hint was to indicate that Jenna should not use argparse for the new stage since the new stage will get its inputs from the pipeline